server/src/main/java/com/ibm/ws/lars/rest/mongo/MongoProducer.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+* Copyright (c) 2018 IBM Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+package com.ibm.ws.lars.rest.mongo;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import com.ibm.websphere.crypto.PasswordUtil;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.WriteConcern;
+import com.mongodb.DB;
+
+import java.util.ArrayList;
+import java.util.Properties;
+import java.util.Vector;
+import java.util.Iterator;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+@ApplicationScoped
+public class MongoProducer {
+    private static final Logger logger = Logger.getLogger(MongoProducer.class.getCanonicalName());
+
+    private String dbName = null;
+
+    @Produces
+    public MongoClient createMongo() {
```

Comment:
As it's written, the `MongoClient` bean is `Dependent` scoped, you'll make a new connection to Mongo every time you create an instance to be injected. If you want to connect once for the whole application, you should add `@ApplicationScoped` to this method as well.

Similarly for `createDB()`.

If you're only injecting `DB` into one place in your app and the thing you're injecting into is application scoped, then it probably won't make any difference, but otherwise you'll be connecting off to MongoDB more than you need to.

Reply:
We tried this - and it does not work because DB does not have a zero-args constructor, and the zero-args constructor for MongoClient tries to connect to the default localhost:27017 mongo server.